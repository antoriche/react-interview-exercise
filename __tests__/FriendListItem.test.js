import React from 'react';
import FriendListItem from '../src/components/FriendListItem';
import renderer from 'react-test-renderer';

describe('FriendListItem', ()=>{
  it('render correctly', () => {
    const component = renderer.create(
      <FriendListItem
          id={0}
          name={"Michel"}
          gender={'M'}
          starred={false}
          starFriend={()=>{}}
      />,
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree).includes("fa-mars")).toBe(true);
    expect(JSON.stringify(tree).includes("fa-star-o")).toBe(true);
  });
  it('render gender', () => {
    const component = renderer.create(
      <FriendListItem
          id={0}
          name={"Michel"}
          gender={'F'}
          starred={false}
          starFriend={()=>{}}
      />,
    );
    
    let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree).includes("fa-venus")).toBe(true);
  });
  it('render starred', () => {
    const component = renderer.create(
      <FriendListItem
          id={0}
          name={"Michel"}
          gender={'M'}
          starred={true}
          starFriend={()=>{}}
      />,
    );
    
    let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree).includes("fa-star-o")).toBe(false);
  });
});