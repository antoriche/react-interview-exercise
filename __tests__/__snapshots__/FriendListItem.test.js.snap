exports[`FriendListItem render correctly 1`] = `
<li
  className={undefined}>
  <div
    className={undefined}>
    <div>
      <i
        className="fa fa-mars" />
      <span
        className={undefined}>
        Michel
      </span>
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div
    className={undefined}>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-star-o" />
    </button>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-trash" />
    </button>
  </div>
</li>
`;
